AAD/UserAADService.cs

```
@@ -0,0 +1,113 @@
+                    .WithAuthority(AzureCloudInstance.AzurePublic, _azureSettings["TenantId"])
+                    .WithCacheOptions(CacheOptions.EnableSharedCacheOptions)
+                    .Build();
+        }
+
+        public async Task<string> GetAccessToken(AuthenticationState authState)
+        {
+            string accessToken = string.Empty;
+
+            //https://database.windows.net/.default
+            var scopes = new string[] { _azureSettings["Scopes"] };
+
+            try
+            {
+                var accountIdentifier = GetAccountIdentifier(authState);
+
+                IAccount account = await _app.GetAccountAsync(accountIdentifier);
+
+                AuthenticationResult authResult = await _app.AcquireTokenSilent(scopes, account).ExecuteAsync();
+                accessToken = authResult.AccessToken;
+            }
+            catch (MsalUiRequiredException)
+            {
+                _consentHandler.ChallengeUser(scopes);
+                return accessToken;
+            }
+
+            return accessToken;
+        }
+
+        public async Task<string> GetDatabaseLoggedUser(AuthenticationState authState)
+        {
+            var loggedUser = "N/A";
+
+            using (SqlConnection conn = _database.GetSqlConnection())
+            {
+                try
+                {
+                    var token = await GetAccessToken(authState);
+
+                    if (string.IsNullOrEmpty(token)) return loggedUser;
+
+                    conn.AccessToken = token;
+
+                    if (conn.State == ConnectionState.Closed)
+                        await conn.OpenAsync();
+
+                    SqlCommand cmd = new(@"SELECT [dbo].[UsernamePrintFn]()", conn);
+
+                    loggedUser = (await cmd.ExecuteScalarAsync()).ToString();
+
+                }
+                catch (Exception)
+                {
+                    return loggedUser;
+                }
```

Would there be any exceptions that would require extra handling?

Not sure if failing silently here could cause issues down the line.